### PR TITLE
HHH-10812 Adding module ZIP and integration test for using Hibernate …

### PR DESCRIPTION
**Pull request by gunnarmorling**
…ORM 5.1 on WildFly 10

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@Sanne Probably you are the best to review this one. Any feedback welcome.

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 1]
```diff
@@ -0,0 +1,54 @@
+= Updating Hibernate ORM within WildFly
+
+The http://wildfly.org/[WildFly application server] comes with Hibernate as the default JPA provider out of the box.
+This means that you don't need to package Hibernate ORM with the applications you deploy on WildFly,
+instead the application server will automatically enable Hibernate support if it detects that your application works with JPA/Hibernate.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: "works with JPA/Hibernate" -> "requires JPA" ?

The "works with" seems a bit ambiguous as we can't really tell how well it works and/or how it's integrated. Also we don't do any kind of detection for "native" Hibernate API usage. I'd rather stick to a more vague "Requires JPA" in the introductory section and then point to specifics further down.

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+There may be times though where a newer version of Hibernate ORM is available than the one coming with a given WildFly release.
+For that case the Hibernate ORM project provides a ZIP file containing the required modules to update a WildFly installation to newer versions.
+The module ZIP file is available from SourceForge and BinTray, alike the ZIP/TAR.GZ distributions (TODO: Add link).
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: TODO ?

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+As of Hibernate ORM 5.1.1, a ZIP file for upgrading WildFly 10 to that Hibernate version (from 5.0.x) is provided.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: s/As of/Since/ to make it a bit future proof ?

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+The general goal is to provide a ZIP file for bringing new versions of Hibernate ORM to the WildFly version current at the time of the Hibernate release.
+
+Once downloaded, extract the contents of the ZIP file into the _modules_ directory of your WildFly installation
+(shut down the application server before, should it be running).
+Note that the Hibernate ORM modules coming with WildFly will remain untouched,
+i.e. you can switch between the original version and the new version from the ZIP file as needed as a matter of configuration.
+
+The module system of the application server uses what's called "slots" to differentiate between several versions of one module.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: N.B. it's correct but the "slots" are going away - see the WildFly-dev discussion. Maybe reword it slightly so you don't have to get back to it in 2 months?

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+By default, slot "main" of the Hibernate ORM modules will be used to provide JPA support for given deployments (representing the Hibernate version coming with WildFly itself).
+In order to use another slot, specify the following two property in the _persistence.xml_ file of your application:
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: let's talk about "module ids" rather than slots (as for the reason above, but also clearer since the properties you're referring to require the full module name)

File: documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+----
+====
+
+For `<%VERSION%>` specify the version of the module ZIP you downloaded, e.g. `5.1.1.Final`.
+Alternatively you can specify just the minor version, e.g. `5.1`.
+That way you can update your WildFly server with further micro updates of the same Hibernate ORM release family as they are released,
+without having to adapt your _persistence.xml_ upon each micro update.
+Note that you can have several micro updates of the same release family next to each other within a WildFly instance
+and switch between them by means of the properties shown above.
+If you are specifying just the minor version as module slot, the modules from the ZIP file you unpacked last will be used.
```
Comment (Sanne) [inline documentation/src/main/asciidoc/topical/wildfly/Wildfly.adoc]: Looks great. Do you think we need to point out that these modules are intended to work with WildFly 10?

Maybe we should clarify that we'll be releasing such "upgrade packs" for selected AS editions, and that people should make sure to take "the right one": not any random combination will necessarily work, nor be an option.

**Comment by gunnarmorling** [comment]
@Sanne, I've pushed one more commit to address your remarks. The ZIP file has the targeted version as a classifier, e.g. _hibernate-orm-modules-5.1.1.Final-wildfly10-dist.zip_. So it should be apparent from that, that this is only meant for a specific WF version.

**Comment by Sanne** [comment]
typo in the JIRA of the commit message ;)

I'll fix it myself.

**Comment by Sanne** [comment]
merged!

I'll create a follow-up JIRA to port it to master (5.2)
